src/crush/CrushWrapper.cc

```diff
@@ -1883,14 +1898,51 @@ int CrushWrapper::device_class_clone(
   if (res)
     return res;
   assert(!bno || bno == *clone);
+
   res = set_item_class(*clone, device_class);
   if (res < 0)
     return res;
+
   // we do not use set_item_name because the name is intentionally invalid
   name_map[*clone] = copy_name;
   if (have_rmaps)
     name_rmap[copy_name] = *clone;
   class_bucket[original_id][device_class] = *clone;
+
+  // set up choose_args for the new bucket. 
+  for (auto w : choose_args) {
+    crush_choose_arg_map cmap = w.second;
+    if (-1-bno >= (int)cmap.size) {
+      unsigned new_size = -1-bno;
```

Comment:
since `-1-bno` works as an subscript, I think it instead should be:

`unsigned new_size = -1-bno + 1;`